sympy/utilities/tests/test_autowrap.py

```diff
@@ -186,6 +192,10 @@ def test_cython_wrapper_compile_flags():
         setup_text = f.read()
     assert setup_text == expected
 
+    os.remove(setup_file_path)
+    os.remove(code_gen.module_name + '.pyx')
+    os.rmdir(temp_dir)
+
```

Comment:
- Why not just delete the entire temp directory at the end of the test and don't worry about all the os.remove() calls?
- I think shutil.rmtree() might be a better function.
- This should be checked on windows before merging.

Reply:
Also, things like this generally go better in a "cleanup" function that runs regardless of test pass/failure.